File: ruby/1-10/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  def self.compute(a, b)
    raise ArgumentError unless a.instance_of?(String) && b.instance_of?(String)
    raise ArgumentError unless a.length == b.length

    distance = 0
    a.chars.each_with_index do |element, index|
      distance += 1 unless element == b[index]
    end
    distance
  end
end

File: go/1-10/hamming/hamming.go
package hamming

import (
	"errors"
)

const testVersion = 5

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("ArgumentError")
	}

	distance := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}

	return distance, nil
}
